entities/itemRepresentation.ts

```diff
@@ -22,13 +24,61 @@ export default class ItemRepresentation extends Phaser.GameObjects.Container {
         if (item.quantity === 1) {
             this.quantityText.setVisible(false);
         }
+        if(scene.scene.key === 'TraderOverlay') {
+            this.priceTag = scene.add.image(0, 23, 'price-tag-icons', 3).setScale(1.8, 1.2)
+            this.coinIcon = scene.add.image(17, 23, 'price-tag-icons', 0).setScale(1.2)
+            this.itemPriceText = scene.add.text(10, 32, '', {
+                font: '14px monospace',
+                color: '#000000',
+                padding: {
+                    left: 1
+                },
+            }).setOrigin(1, 1);
+        }
+        this.add([this.priceTag])
+        this.add([this.itemPriceText])
+        this.add([this.coinIcon])
         this.setSize(64, 64)
             .setScrollFactor(0)
             .setInteractive();
 
         scene.add.existing(this);
     }
 
+    public setPriceTag(characterMoney: number, forWho: String){
+        if (forWho === "player") {
+          this.itemPriceText.setText(this.item.buyPrice.toString());
+          this.setPriceTagColor(characterMoney >= this.item.buyPrice ? 'green' : 'red');
+        } else if (forWho === "npc") {
+          if (this.item.sellPrice === 0) {
+            this.hidePriceInformation();
+          } else {
+            this.itemPriceText.setText(this.item.sellPrice.toString());
+            this.setPriceTagColor(characterMoney >= this.item.sellPrice ? "green" : "red");
+          }
+        } else {
+          throw Error("invalid argument!");
+        }
+    }
+
+    public setPriceTagColor(color: String){
```

Comment:
Here, for typing, you can use 
public setPriceTagColor(color: 'green' | 'red'){ ...